pages/docs/with-nextjs.en-US.md

```diff
@@ -26,13 +26,13 @@ You can pass the pre-fetched data as the initial value to the `initialData` opti
  export async function getStaticProps() {
   // `getStaticProps` is invoked on the server-side,
   // so this `fetcher` function will be executed on the server-side.
-  const posts = await fetcher('/api/posts')
+  const posts = await fetcher('https://jsonplaceholder.typicode.com/posts')
   return { props: { posts } }
 }
 
 function Posts (props) {
   // Here the `fetcher` function will be executed on the client-side.
-  const { data } = useSWR('/api/posts', fetcher, { initialData: props.posts })
+  const { data } = useSWR('https://jsonplaceholder.typicode.com/posts', fetcher, { initialData: props.posts })
```

Comment:
```suggestion
  const { data } = useSWR('/api/posts', fetcher, { initialData: props.posts })
```


for requests inside component, this will result in cross origin issues. IMO we can do:

for SSG: use 3rd party link for getting posts
for SSR: still use API endpoint, and serverless API will return the response from 3rd party endpoint

Reply:
The context of the document section was, with SWR we can prerender the page, by fetching data at build time(SSG) or server side(SSR) and also use SWR using the `initialData` option of `useSWR()`. To my knowledge, we cannot fetch from an API route inside `getStaticProps` or `getServerSideProps`, instead we can implement the same logic inside an API route into the `getStaticProps`.

I agree with the CORS issue.

I have another idea, how about i replace the `fetcher('/api/posts')` with a dummy API logic in `getStaticProps`?
That would resolve the issue of fetching from API route and also implementing the `initialData` option.
What do you say?